os/net/mac/ble/ble-l2cap.c

```diff
@@ -437,6 +437,12 @@ input_l2cap_frame_flow_channel(l2cap_channel_t *channel, uint8_t *data, uint16_t
 
     memcpy(&channel->rx_buffer.sdu_length, &data[4], 2);
 
+    if (payload_len > data_len - 6) {
```

Comment:
Can you join this if statement with the condition on line 432? They are both checking payload_len.

Also, you can repeat for the else part of the branch.

Reply:
Addressed in latest commit